src/main/java/org/jetbrains/research/intellijdeodorant/ide/refactoring/ReplaceTypeCodeWithStateStrategy.java

```diff
@@ -0,0 +1,2164 @@
+                        break;
+                    }
+                }
+            }
+        } else {
+            String packageName = PsiUtil.getPackageName(sourceTypeDeclaration);
+            if (packageName != null && !packageName.isEmpty()) {
+                PsiPackageStatement packageStatement = elementFactory.createPackageStatement(packageName);
+                intermediateClassFile.add(packageStatement);
+            }
+            intermediateClassTypeDeclaration = elementFactory.createClass(intermediateClassName);
+            intermediateClassTypeDeclaration.getExtendsList().add(elementFactory.createReferenceFromText(abstractClassName, null));
+            PsiUtil.setModifierProperty(intermediateClassTypeDeclaration, PsiModifier.PUBLIC, true);
+            PsiUtil.setModifierProperty(intermediateClassTypeDeclaration, PsiModifier.ABSTRACT, true);
+        }
+
+        PsiType concreteMethodReturnType;
+        if (returnedVariable == null && !typeCheckElimination.typeCheckCodeFragmentContainsReturnStatement()) {
+            concreteMethodReturnType = PsiType.VOID;
+        } else {
+            if (returnedVariable != null) {
+                concreteMethodReturnType = returnedVariable.getType();
+            } else {
+                concreteMethodReturnType = typeCheckElimination.getTypeCheckMethodReturnType();
+            }
+        }
+        PsiMethod concreteMethodDeclaration = elementFactory.createMethod(typeCheckElimination.getAbstractMethodName(), concreteMethodReturnType);
+        PsiUtil.setModifierProperty(concreteMethodDeclaration, PsiModifier.PUBLIC, true);
+
+        PsiParameterList concreteMethodParametersRewrite = concreteMethodDeclaration.getParameterList();
+        if (returnedVariable != null) {
+            PsiParameter parameter = elementFactory.createParameter(returnedVariable.getName(), returnedVariable.getType());
+            concreteMethodParametersRewrite.add(parameter);
+        }
+        for (PsiParameter abstractMethodParameter : typeCheckElimination.getAccessedParameters()) {
+            if (!abstractMethodParameter.equals(returnedVariable)) {
+                concreteMethodParametersRewrite.add(abstractMethodParameter);
+            }
+        }
+        for (PsiVariable fragment : typeCheckElimination.getAccessedLocalVariables()) {
+            if (!fragment.equals(returnedVariable)) {
+                PsiParameter parameter = elementFactory.createParameter(fragment.getName(), fragment.getType());
+                concreteMethodParametersRewrite.add(parameter);
+            }
+        }
+
+        if (sourceTypeRequiredForExtraction()) {
+            PsiType parameterType = PsiTypesUtil.getClassType(sourceTypeDeclaration);
+            String parameterName = sourceTypeDeclaration.getName();
+            parameterName = parameterName.substring(0, 1).toLowerCase() + parameterName.substring(1);
+            concreteMethodParametersRewrite.add(elementFactory.createParamete
```

Comment:
?

Reply:
Looks like this comment and the two ones above are broken -- they are pointing to the same line. Was this comment about something else?